hail/src/main/scala/is/hail/expr/types/physical/PNDArray.scala

```diff
@@ -8,6 +8,10 @@ import is.hail.expr.types.virtual.TNDArray
 import is.hail.utils._
 import is.hail.asm4s._
 
+final class StaticallyKnownField[T, U](
```

Comment:
Only used in PNDArray

Reply:
seems like a reasonably nice codegen primitive, but fine to move it here.

Reply:
Sure, without guidance on its future uses its hard to understand why it's located here. If we can document that somewhere, I'm happy to move back.